assets/js/modules/reader-revenue-manager/components/dashboard/ReaderRevenueManagerSetupCTABanner.js

```diff
@@ -37,268 +35,164 @@ import { __ } from '@wordpress/i18n';
+	const { triggerSurvey } = useDispatch( CORE_USER );
 
-	const isDismissed = useSelect( ( select ) =>
-		select( CORE_USER ).isPromptDismissed(
-			READER_REVENUE_MANAGER_SETUP_BANNER_DISMISSED_KEY
-		)
-	);
+	const readerRevenueManagerDocumentationURL =
+		'https://readerrevenue.withgoogle.com';
 
-	const dismissCount = useSelect( ( select ) =>
-		select( CORE_USER ).getPromptDismissCount(
-			READER_REVENUE_MANAGER_SETUP_BANNER_DISMISSED_KEY
-		)
+	const isDismissalFinal = useSelect( ( select ) =>
+		select( CORE_NOTIFICATIONS ).isNotificationDismissalFinal( id )
 	);
 
+	// See TODO note below.
+	const isCTADismissed = useSelect( ( select ) =>
+		select( CORE_NOTIFICATIONS ).isNotificationDismissed( id )
+	);
 	const dismissedPromptsLoaded = useSelect( ( select ) =>
 		select( CORE_USER ).hasFinishedResolution( 'getDismissedPrompts', [] )
 	);
-
-	const { dismissPrompt, triggerSurvey } = useDispatch( CORE_USER );
-
-	const onDismiss = useCallback( () => {
-		trackEvent(
-			`${ viewContext }_rrm-setup-notification`,
-			'dismiss_notification'
-		).finally( () => {
-			const expirationInSeconds =
-				dismissCount < 1 ? 2 * WEEK_IN_SECONDS : 0;
-
-			showTooltip();
-
-			dismissPrompt( READER_REVENUE_MANAGER_SETUP_BANNER_DISMISSED_KEY, {
-				expiresInSeconds: expirationInSeconds,
-			} );
-		} );
-	}, [ dismissCount, dismissPrompt, showTooltip, viewContext ] );
-
-	const readerRevenueManagerDocumentationURL =
-		'https://readerrevenue.withgoogle.com';
-
-	const canActivateRRMModule = useSelect( ( select ) =>
-		select( CORE_MODULES ).canActivateModule(
-			READER_REVENUE_MANAGER_MODULE_SLUG
-		)
-	);
-
-	const showBanner =
-		isDismissed === false &&
-		canActivateRRMModule &&
-		dismissedPromptsLoaded === true;
+	const hideCTABanner = isCTADismissed || ! dismissedPromptsLoaded;
 
 	useEffect( () => {
-		if ( showBanner ) {
-			trackEvent(
-				`${ viewContext }_rrm-setup-notification`,
-				'view_notification'
-			);
-
+		if ( ! hideCTABanner ) {
 			triggerSurvey( 'view_reader_revenue_manager_cta' );
 		}
-	}, [ showBanner, triggerSurvey, viewContext ] );
+	}, [ hideCTABanner, triggerSurvey ] );
 
 	if ( isTooltipVisible ) {
 		return (
 			<Fragment>
-				<WidgetNull />
 				<AdminMenuTooltip
 					title=""
 					content={ __(
 						'You can always enable Reader Revenue Manager from Settings later',
 						'google-site-kit'
 					) }
 					dismissLabel={ __( 'Got it', 'google-site-kit' ) }
-					tooltipStateKey={
-						READER_REVENUE_MANAGER_SETUP_BANNER_DISMISSED_KEY
-					}
+					tooltipStateKey={ id }
 				/>
 			</Fragment>
 		);
 	}
 
-	if ( ! showBanner ) {
-		return <WidgetNull />;
+	// TODO Remove this hack
+	// We "incorrectly" pass true to the `skipHidingFromQueue` option when dismissing this banner.
+	// This is because we don't want the component removed from the DOM as we have to still render
+	// the `AdminMenuTooltip` in this component. This means that we have to rely on manually
+	// checking for the dismissal state here.
```

Comment:
Do we have an issue for this? We should address this sooner than later and then remove the `skipHidingFromQueue` option as it shouldn't be needed, correct?

Reply:
Correct, I've added #10003 to tackle this. Thanks.